### PR TITLE
ip, w3emc, and misc. CI updates

### LINKED ISSUE #147: build breaks for latest ip library release, needs use ip_mod
We need to update to handle the latest ip release.

```
cd /home/ed/NCEPLIBS-grib_util/b/src/copygb && /home/ed/.local/lib/python3.8/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/copygb.dir/link.txt --verbose=1
/usr/bin/gfortran -g -fdefault-real-8  -O3 -g -fdefault-real-8  CMakeFiles/copygb.dir/copygb.F90.o -o copygb  /usr/local/NCEPLIBS-bacio-2.5.0/lib/libbacio.a /usr/local/NCEPLIBS-w3emc-2.9.2/lib/libw3emc_d.a /usr/local/NCEPLIBS-ip-4.0.0/lib/libip_d.a /usr/local/NCEPLIBS-sp-2.3.3/lib/libsp_d.a /usr/local/NCEPLIBS-bacio-2.5.0/lib/libbacio.a 
/usr/bin/ld: CMakeFiles/copygb.dir/copygb.F90.o: in function `intgrib1_':
/home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1222: undefined reference to `ipolates_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1297: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1328: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1287: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1228: undefined reference to `ipolates_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1265: undefined reference to `ipolatev_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1250: undefined reference to `ipolates_'
/usr/bin/ld: /home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:1209: undefined reference to `ipolates_'
/usr/bin/ld: CMakeFiles/copygb.dir/copygb.F90.o: in function `MAIN__':
/home/ed/NCEPLIBS-grib_util/src/copygb/copygb.F90:257: undefined reference to `makgds_'
collect2: error: ld returned 1 exit status
make[2]: *** [src/copygb/CMakeFiles/copygb.dir/build.make:101: src/copygb/copygb] Error 1
make[2]: Leaving directory '/home/ed/NCEPLIBS-grib_util/b'
make[1]: *** [CMakeFiles/Makefile2:1075: src/copygb/CMakeFiles/copygb.dir/all] Error 2
make[1]: Leaving directory '/home/ed/NCEPLIBS-grib_util/b'
make: *** [Makefile:146: all] Error 2

```

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR consists of 6 related commits:
 - Clean up spack CI (e753b91ebee812664907a44e9b43214090f0229b)
 - Reinstate copygb tests (which includes testing with ip; 3f38d39eca48505e06cd6925137420aefb1c7297)
 - Add copygb2 testing that uses interpolation (6e2c4236cd8129f431421472c9cde7bdfdbab7ef; 55639b6e8a3fe2d3d04be8386837215d9f6f9c8b)
 - Move to w3emc 2.10.0 (4a9b2905eff517954e238994bcf24d5845c79150)
 - Update to ip v4 (b42ed510f179540b14822643693944093cc1e028)

This PR replaces #242. It is waiting on https://github.com/spack/spack/pull/39075 which will make w3emc@2.10.0 available through Spack in order for the spack.yml workflow to succeed (I have tested it successfully with my own Spack branch so the rest will run through).

We also need to update the branch protection rules so that the now-deprecated w3emc/2.9.3 Linux_versions CI tests are not required for merging.

Fixes #147, #93, #244, #245
Addresses #116

### CODE REVIEW

File: .github/workflows/Intel.yml [hunk 1]
```diff
@@ -196,7 +196,7 @@ jobs:
       run: |
         cd grib_utils
         mkdir build && cd build
-        cmake -DCMAKE_PREFIX_PATH="~/bacio;~/jasper;~/sp;~/ip;~/w3emc;~/g2" ..
+        cmake -DCMAKE_PREFIX_PATH="~/bacio;~/jasper;~/sp;~/ip;~/w3emc;~/g2" -DCMAKE_Fortran_FLAGS="-heap-arrays" ..
```
Comment (edwardhartnett) [inline .github/workflows/Intel.yml:199]: What's with this new fortran flag?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -92,10 +93,12 @@ gu_test(run_degrib2_tests)
 gu_test(run_grbindex_tests)
 gu_test(run_grb2index_tests)
 
-# This test depends on the g2c_compare utility, which may not be
-# present.
-if (G2C_COMPARE)
+option(G2C_COMPARE OFF)
```
Comment (edwardhartnett) [inline tests/CMakeLists.txt:96]: Can you put this option in the top-level cmake file, with all the other build options?

**Review by edwardhartnett — APPROVED** [review]

File: .github/workflows/Intel.yml (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply .github/workflows/Intel.yml:199]: I stole this from the ip tests to resolve the segfault issues with Intel. I've run similar situations where the compiler seems to be cramming large temporary arrays into stack memory and then segfaulting, though I admit I still don't fully understand the underlying problem.

**Comment by AlexanderRichert-NOAA** [comment]
The spack PR got merged with lightning speed, so all we need to do is update the branch protection settings for the two Linux_versions tests with w3emc 2.9.3 (which I do not have permissions to do).

**Comment by edwardhartnett** [comment]
I have added you as an admin of this repo so you can change the branch protection rules. Let me know if you need help, and take care!

**Review by GeorgeGayno-NOAA — APPROVED** [review]

File: src/copygb2/copygb2.F90 [hunk 3]
```diff
@@ -1349,6 +1349,7 @@ END SUBROUTINE INTGRIB2
   SUBROUTINE INTGRIB1(K1F,KGDS1F,K2F,KGDS2F,MRL,MRO, &
        IV,IP,IPOPT,KGDS1,K1,IB1,L1,F1,G1,KGDS2,K2, &
        IB2,L2,F2,G2,IRET)
+    USE IP_MOD
```
Comment (edwardhartnett) [inline src/copygb2/copygb2.F90:1352]: OK, last night I realized that these "USE IP_MOD" statements should be wrapped in an #ifdef which checks whether ip-4.0.0 or greater was found.

In this way, the code would continue to work with older versions if ip.

**Pull request merged**